src/App.js

```diff
@@ -26,43 +27,72 @@ const generateSquares = () => {
 }
 
 const App = () => {
-
-  // This starts state off as a 2D array of JS objects with
-  // empty value and unique ids.
+  const [player, setPlayer] = useState(PLAYER_1);
   const [squares, setSquares] = useState(generateSquares());
+  const [winner, setWinner] = useState('')
+
+  const onClickCallback = (squareID, value) => {
+  
+    if (winner || value) return;
+    const updatedSquares = []
+    for (let i = 0; i < 3; i++) {
+      let row = []
+      for (let j = 0; j < 3; j++) {
+        (squares[i][j].id === squareID) ? row.push({id: squareID, value: player}) : row.push(squares[i][j])
+      } 
+      updatedSquares.push(row)
+    }
+    (player === PLAYER_1) ? setPlayer(PLAYER_2) : setPlayer(PLAYER_1);
+    setSquares(updatedSquares);
+    checkForWinner(updatedSquares);
+  };
 
-  // Wave 2
-  // You will need to create a method to change the square 
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
 
+  const checkForWinner = (squares) => {
+    const checkForThree = (first, second, third) => {
```

Comment:
Nice work writing a helper function for a clear and readable algorithm that works with a 2D array.